tests/conftest.py

```
@@ -68,90 +72,136 @@ def oauth2_token_dict() -> dict:
+    remaining = list(interactions)
+    lock = threading.Lock()
+
+    def mock_request(
+        self,
+        method,
+        subdomain,
+        path,
+        /,
+        api=False,
+        referrer=False,
+        headers={},
```

Ruff (0.15.7)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)